src/main/java/uk/gov/hmcts/reform/locationrefdata/camel/processor/CourtVenueProcessor.java

```diff
@@ -87,14 +98,42 @@ public void process(Exchange exchange) throws Exception {
         }
 
         if (filteredCourtVenues.size() != jsrValidatedCourtVenues) {
-            setFileStatus(exchange, applicationContext);
+            setFileStatus(exchange, applicationContext,PARTIAL_SUCCESS);
+        }
+
+        if (courtVenues.size() > 0) {
+            processExceptionRecords(exchange, courtVenues);
         }
 
         exchange.getMessage().setBody(filteredCourtVenues);
+
+    }
+
+
+    private void processExceptionRecords(Exchange exchange,
+                                         List<CourtVenue> courtVenuesList) {
+
+        List<Pair<String, Long>> zeroByteCharacterRecords = new ArrayList<>();
+        courtVenuesList.forEach(courtVenue -> dataQualityCheckConfiguration.zeroByteCharacters
```

Comment:
public field dereference, please consider defining method for zeroByteCharacters and accessing via method

Reply:
zeroBytesCharacters is still being accessed outside of object directly. please introduce 'getter' method and access via method

Reply:
added getter setters